### PR TITLE
PDE-2872 fix(legacy-scripting-runner): `z.dehydrateFile(url, {})` should not send auth

### PR DESCRIPTION
**Pull request by eliangcs**
## Legacy scripting code to reproduce the issue

```js
var Zap = {
  email_catch_hook: {
    bundle.cleaned_request.file = z.dehydrateFile('https://httpbin.zapier-tooling.com/get', {});
    return bundle.cleaned_request;
  }
};
```

## Expected behavior

When hydrated, we should NOT see the auth headers or params in the request.

But if you do `z.dehydrateFile('https://httpbin.zapier-tooling.com/get')` (omitting the second argument for request options), we _should_ see the auth headers or params.

## Current behavior

When hydrated, we see the auth headers or params in the request. Some server doesn't like it when the URL you're requesting for doesn't require auth but you give it auth info. For example, AWS S3 would give you HTTP 400 error saying `Unsupported Authorization Type`.

### CODE REVIEW

**Comment by eliangcs** [comment]
@xavdid let's ship this together with your PR https://github.com/zapier/zapier-platform/pull/444 in version 3.8.4. I can do it on Monday if you don't do it on your Friday.

File: packages/legacy-scripting-runner/index.js
Comment (xavdid) [inline packages/legacy-scripting-runner/index.js:1397]: nit, but you can do `bundle.inputData.request == null` (double eq), which covers both `null` and `undefined`

**Review by xavdid — APPROVED** [review]
Left a comment, but it's not worth fixing and having to wait for another review. So think of it as an FYI.

**Comment by xavdid** [comment]
oh, and i've got a short friday so I'll let you handle the release. I added your PR to the changelog in my branch, so you should be able to approve+merge mine, merge yours, and do the release. I can also check in during my sunday evening to clear any blockers so this gets out on monday

File: packages/legacy-scripting-runner/index.js
Comment (eliangcs) [reply packages/legacy-scripting-runner/index.js:1397]: I'll merge this PR as-is because I don't want to wait for another review. But that's another "good" thing to know about JavaScript. Thanks!

**Pull request merged**
